File: app/src/main/java/com/bms/rabbit/features/auth/AuthNetworkDataSource.java
package com.bms.rabbit.features.auth;
// Created by Konstantin on 13.08.2018.


import com.bms.rabbit.entities.NewUser;
import com.bms.rabbit.entities.UserResponse;

import java.util.HashMap;
import java.util.Map;

import io.reactivex.Single;
import retrofit2.Retrofit;

public class AuthNetworkDataSource {
    private AuthApi authApi;

    public AuthNetworkDataSource(Retrofit retrofit) {
        authApi = retrofit.create(AuthApi.class);
    }

    public Single<UserResponse> createUser(NewUser newUser) {
        Map<String,String> fieldMap = new HashMap<>();
        fieldMap.put("name",newUser.getName());
        fieldMap.put("email",newUser.getMail());
        fieldMap.put("group_code",newUser.getCode());
        return authApi.createUser(fieldMap);
    }
}

File: app/src/main/java/com/bms/rabbit/features/auth/AuthNetworkDataSource.kt
package com.bms.rabbit.features.auth

// Created by Konstantin on 13.08.2018.


import com.bms.rabbit.entities.NewUser
import com.bms.rabbit.entities.UserResponse

import java.util.HashMap

import io.reactivex.Single
import retrofit2.Retrofit

class AuthNetworkDataSource(retrofit: Retrofit) {
    private val authApi: AuthApi = retrofit.create(AuthApi::class.java)

    fun createUser(newUser: NewUser): Single<UserResponse> {
        val fieldMap = HashMap<String, String>()
        fieldMap["name"] = newUser.name
        fieldMap["email"] = newUser.mail
        fieldMap["group_code"] = newUser.code
        fieldMap["client_key"] = newUser.clientKey
        return authApi.createUser(fieldMap)
    }
}
